src/middlewared/middlewared/plugins/jbof/crud.py

```diff
@@ -0,0 +1,219 @@
+import middlewared.sqlalchemy as sa
+from middlewared.schema import accepts, Dict, Int, IPAddr, Password, Patch, Str
+from middlewared.service import CRUDService, private, ValidationErrors
+
+from middlewared.plugins.jbof.redfish import RedfishClient, InvalidCredentialsError
+
+
+class JBOFModel(sa.Model):
+    __tablename__ = 'storage_jbof'
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+
+    jbof_description = sa.Column(sa.String(120), nullable=True)
+
+    # When performing static (code-based) assignment of data-plane IPs, we
+    # want each JBOD to have a deterministic unique index that counts up from
+    # zero (without any gaps, which rules out using the id).  This will not be
+    # part of the public API.
+    jbof_index = sa.Column(sa.Integer(), unique=True)
+    jbof_uuid = sa.Column(sa.Text(), nullable=False, unique=True)
+
+    jbof_mgmt_ip1 = sa.Column(sa.String(45), nullable=False)
+    jbof_mgmt_ip2 = sa.Column(sa.String(45), default='')
+    jbof_mgmt_username = sa.Column(sa.String(120))
+    jbof_mgmt_password = sa.Column(sa.EncryptedText())
+
+
+class JBOFService(CRUDService):
+
+    class Config:
+        service = 'jbof'
+        datastore = 'storage.jbof'
+        datastore_prefix = "jbof_"
+        cli_private = True
+
+    ENTRY = Dict(
+        'jbof_entry',
+        Int('id', required=True),
+
+        Str('description'),
+
+        # Redfish
+        IPAddr('mgmt_ip1', required=True),
+        IPAddr('mgmt_ip2', required=False),
+        Str('mgmt_username', required=True),
+        Password('mgmt_password', required=True),
```

Comment:
Not saying that this is a problem, just noting for reference that you will need to possibly think through behavior of when user restores server configuration from backup without the 'secret seed'. In this case mgmt_password will be empty string.